tcms/core/widgets.py

```diff
@@ -40,3 +41,18 @@ class Media:
             "prismjs/plugins/autoloader/prism-autoloader.min.js",
             "js/simplemde_security_override.js",
         ]
+
+
+class DurationWidget(forms.Textarea):
+
+    def index(request, attrs=None):
```

Comment:
That was previously the `render()` method. IDK what `index()` does. Please explain!

Reply:
I used the index() method after seeing the documentation on how to[ render HTML from a template](https://docs.djangoproject.com/en/3.2/intro/tutorial03/) which you linked before but I noticed that its often used int he views.py class instead of widgets.py. 

Is it possible for me to delete the template with the duration I made before and instead render the HTML from the custom widget like it did before? Because it works when I render html from the custom widget but maybe this is not good in the long term?

Reply:
> @atodorov I used the index() method after seeing the documentation on how to[ render HTML from a template](https://docs.djangoproject.com/en/3.2/intro/tutorial03/) which you linked before but I noticed that its often used int the views.py file instead of widgets.py.

In that context `index` is the name of the view function which in the tutorial is supposed to render the home page. In the context of a custom widget we're dealing with classes which have their methods (aka their interface) defined by the parent class and by the rest of the Django framework.  You can't just change them without actually verifying the behavior. At present even the slightly broken behavior from https://github.com/kiwitcms/Kiwi/pull/2314#issuecomment-820864916 is not present and it is actually worse.

> 
> Is it possible for me to delete the template with the duration and instead render the HTML from the custom widget like it did before? Because it works when I render html from the custom widget